Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3493,6 +3496,17 @@ private unsafe void OnUpdate(InputUpdateType updateType, ref InputEventBuffer ev
             m_CurrentUpdate = default;
         }
 
+        private void LogProcessedEventForDevice(InputDevice device, uint totalEventBytesProcessed, InputEventPtr eventPtr)
+        {
+            if (m_Settings.debugLogProcessedEventForDevice && (m_CurrentUpdate != InputUpdateType.Editor))
```

Comment:
I would move this if statement into the outer if statement to avoid doing a potentially non-inlined function call for just doing an early return.

Reply:
I wonder if this couldn't be replaced with a callback instead and let a Debug.Log implementation of a handler exist that could be added by a corresponding setting. This would allow other subscribers as well, e.g. Debugger window, statistical computations etc... I wonder if this isn't already covered somehow, might need to checking into how recording system or debug window does similar things.